server/src/main/java/org/opensearch/cluster/decommission/DecommissionAttribute.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public final class DecommissionAttribute implements Writeable {
+    private final String attributeName;
+    private final List<String> attributeValues;
+
+    /**
+     * Update the attribute values for a given attribute name to decommission
+     *
+     * @param decommissionAttribute current decommissioned attribute object
+     * @param attributeValues       values to be updated with
+     */
+    public DecommissionAttribute(DecommissionAttribute decommissionAttribute, List<String> attributeValues) {
+        this(decommissionAttribute.attributeName, attributeValues);
+    }
+
```

Comment:
Looks like a copy constructor, do we need to copy/clone instances?

Reply:
Removed this